go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -175,25 +179,69 @@ func (oc *DefaultNetworkController) syncEgressFirewall(egressFirewalls []interfa
 		}
 		existingEFNamespaces[ef.Namespace] = true
 	}
+	// move ACLs from cluster port group to per-namespace port group
+	// old acl matches on the address set, which is still valid. match change form address set to port group will be
+	// performed by the egress firewall handler.
+	// sync the ovn and k8s egressFirewall states.
 	predicateIDs := libovsdbops.NewDbObjectIDs(libovsdbops.ACLEgressFirewall, oc.controllerName, nil)
 	aclP := libovsdbops.GetPredicate[*nbdb.ACL](predicateIDs, nil)
 	efACLs, err := libovsdbops.FindACLsWithPredicate(oc.nbClient, aclP)
 	if err != nil {
 		return fmt.Errorf("cannot find Egress Firewall ACLs: %v", err)
 	}
-	deleteACLs := []*nbdb.ACL{}
-	for _, efACL := range efACLs {
-		if !existingEFNamespaces[efACL.ExternalIDs[libovsdbops.ObjectNameKey.String()]] {
-			// there is no egress firewall in that namespace, delete
-			deleteACLs = append(deleteACLs, efACL)
-		}
+	var nsACLs = map[string][]*nbdb.ACL{}
+
+	for _, acl := range efACLs {
+		namespace := acl.ExternalIDs[libovsdbops.ObjectNameKey.String()]
+		// no key will be treated as empty namespace and just deleted
+		nsACLs[namespace] = append(nsACLs[namespace], acl)
 	}
+	batchCounter := 0
+	var ops []libovsdb.Operation
 
-	err = libovsdbops.DeleteACLsFromPortGroups(oc.nbClient, []string{types.ClusterPortGroupNameBase}, deleteACLs...)
-	if err != nil {
-		return err
+	for namespace, acls := range nsACLs {
```

Comment:
Wouldn't it make sense to have another generic `Batch` utility function that is able to batch across mapped values? As a bonus, you would be able to test it specifically.

Reply:
hehe I thought initially that this functionality won't be very popular, but it looks prettier, so worth it :D